src/algebra/big_operators.lean

```diff
@@ -827,6 +857,23 @@ end
 
 end linear_ordered_comm_ring
 
+section canonically_ordered_comm_semiring
+
+variables [decidable_eq α] [canonically_ordered_comm_semiring β]
+
+lemma prod_le_prod' {s : finset α} {f g : α → β} (h : ∀ i ∈ s, f i ≤ g i) :
```

Comment:
`to_additive`?
Also, isn't there already a `prod_le_prod`? How is this one different?

Reply:
It's a typeclass thing: there is no version yet that works on `canonically_ordered_comm_semiring β` (a.k.a. `nnreal` for my purposes). And there is no analogue of this in the additive world, as far as I can tell.

Reply:
The other `prod_le_prod` has a comment 
`/- this is also true for a ordered commutative multiplicative monoid -/`.
Is there actually a common generalisation?